Makefile

```diff
@@ -41,6 +41,15 @@ tests:
 	@command -v helm >/dev/null 2>&1 || { echo "helm is required but not installed"; exit 1; }
 	@helm unittest charts/eoapi -f 'tests/*.yaml' -v charts/eoapi/test-helm-values.yaml
 
+lint:
+	@echo "Running linting and code quality checks..."
+	@if [ ! -f .git/hooks/pre-commit ]; then \
+		echo "Installing pre-commit..."; \
+		uv pip install --system pre-commit yamllint shellcheck-py || pip3 install --user pre-commit yamllint shellcheck-py; \
```

Comment:
Unsure on the `--system` usage, will this install `pre-commit` et al on my system python? I probably don't want that.